src/dev/run_i18n_extract.ts

```diff
@@ -59,19 +59,19 @@ run(
     }
     const srcPaths = Array().concat(path || ['./src', './packages']);
 
-    const list = new Listr([
+    const list = new Listr<ListrContext>([
       {
         title: 'Merging .i18nrc.json files',
         task: () => new Listr(mergeConfigs(includeConfig), { exitOnError: true }),
       },
       {
         title: 'Extracting Default Messages',
         task: ({ config }) =>
-          new Listr(extractDefaultMessages(config, srcPaths), { exitOnError: true }),
+          new Listr(extractDefaultMessages(config!, srcPaths), { exitOnError: true }),
```

Comment:
```suggestion
          new Listr(extractDefaultMessages(config, srcPaths), { exitOnError: true }),
```

update `extractDefaultMessages` to accept an optional config just like `checkCompatibility`. This should never happen based on the order that they are called in, but should the order ever be swapped, the error will point us to the problem